kahypar/utils/validate.h

```diff
@@ -0,0 +1,305 @@
+    } else if (hyperedge_indices[he + 1] - hyperedge_indices[he] <= 1) {
+      report_error(InputErrorType::HyperedgeSizeZeroOrOne, he);
+      ignore_hyperedge = true;
+    } else if (hyperedge_indices[he + 1] - hyperedge_indices[he] > num_hypernodes) {
+      // this check should hopefully prevent an overly long running time in case the indices are garbage
+      report_error(InputErrorType::HyperedgeOutOfBounds, he);
+      continue;
+    }
+    // check hyperedge_weight
+    if (hyperedge_weights != nullptr && hyperedge_weights[he] == 0) {
+      report_error(InputErrorType::HyperedgeWeightZero, he);
+      ignore_hyperedge = true;
+    }
+    if (ignore_hyperedge && ignored_hes != nullptr) {
+      ignored_hes->push_back(he);
+    }
+
+    // check pins
+    pin_set.reset();
+    bool invalid_pin = false;
+    bool duplicate_pin = false;
+    for (size_t i = hyperedge_indices[he]; i < hyperedge_indices[he + 1]; ++i) {
+      const HypernodeID hn = hyperedges[i];
+      bool ignore_pin = ignore_hyperedge; // all pins of an ignored hyperedge are also ignored
+      if (hn >= num_hypernodes) {
+        invalid_pin = true;
+      } else if (pin_set[hn]) {
+        duplicate_pin = true;
+        ignore_pin = true; // ignore duplicate pins
+      } else {
+        pin_set.set(hn);
+      }
+      if (ignore_pin && ignored_pins != nullptr) {
+        ignored_pins->push_back(i);
+      }
+    }
+    // Note: We report invalid/duplicate pins once per hyperedge to avoid emitting
+    // multiple errors for one line of the input file
+    if (invalid_pin) {
+      report_error(InputErrorType::HyperedgeInvalidPin, he);
+    }
+    if (duplicate_pin) {
+      report_error(InputErrorType::HyperedgeDuplicatePin, he);
+    }
+  }
+  if (vertex_weights != nullptr) {
+    for (HypernodeID hn = 0; hn < num_hypernodes; ++hn) {
+      if (vertex_weights[hn] == 0) {
+        report_error(InputErrorType::HypernodeWeightZero, hn);
+      }
+    }
+  }
+  return found_error;
+}
+
+
+/*!
+* Prints the list of errors, including line numbers if provided.
+*
+* \param num_hypernodes Number of hypernodes |V|
+* \param num_hyperedges Number of hyperedges |E|
+* \param errors List of errors from call to validateInput
+* \param line_numbers Mapping of hyperedge/vertex IDs to line of input file.
+*                     Must either be empty or have size num_hyperedges + num_hypernodes
+* \param promote_warnings_to_errors If true, non-fatal errors are reported as error instead of warning
+* \param out Output stream (default: stderr)
+*/
+void printErrors(const HypernodeID num_hypernodes, const HyperedgeID num_hyperedges,
+                 const std::vector<InputError>& errors,
+                 const std::vector<size_t>& line_numbers = {},
+                 bool promote_warnings_to_errors = false,
+                 std::ostream& out = std::cerr) {
+  ASSERT(line_numbers.empty() || line_numbers.size() >= static_cast<size_t>(num_hyperedges));
+  unused(num_hypernodes);
```

Comment:
Why do we need to pass in `num_hypernodes`?

Reply:
Yeah, I originally included the parameter for symmetry with other functions, but it is indeed not needed in this case. Is removed now